src/finbars/finbars.ts

```diff
@@ -0,0 +1,870 @@
+     *
+     * This convenience function is an programmatic alternative to hardcoding the correct style with the correct value in your stylesheet; or setting the correct style with the correct value in the {@link FinBar#barStyles|barStyles} object.
+     *
+     * @see {@link FinBar#foreshortenBy|foreshortenBy}.
+     *
+     * @param {FinBar|null} otherFinBar - Other scrollbar to avoid by shortening this one; `null` removes the trailing space
+     * @returns {FinBar} For chaining
+     */
+    shortenBy: function (otherFinBar) { return this.shortenEndBy('trailing', otherFinBar); },
+
+    /**
+     * @summary Shorten leading end of scrollbar by thickness of some other scrollbar.
+     * @desc Supports non-classical scrollbar scenarios where vertical scroll bar may be on left and horizontal scrollbar may be on top, in which case you want to shorten the "leading end" rather than the trailing end.
+     * @see {@link FinBar#shortenBy|shortenBy}.
+     * @param {FinBar|null} otherFinBar - Other scrollbar to avoid by shortening this one; `null` removes the trailing space
+     * @returns {FinBar} For chaining
+     */
+    foreshortenBy: function (otherFinBar) { return this.shortenEndBy('leading', otherFinBar); },
+
+    /**
+     * @summary Generalized shortening function.
+     * @see {@link FinBar#shortenBy|shortenBy}.
+     * @see {@link FinBar#foreshortenBy|foreshortenBy}.
+     * @param {string} whichEnd - a CSS style property name or an orientation hash name that translates to a CSS style property name.
+     * @param {FinBar|null} otherFinBar - Other scrollbar to avoid by shortening this one; `null` removes the trailing space
+     * @returns {FinBar} For chaining
+     */
+    shortenEndBy: function (whichEnd, otherFinBar) {
+        if (!otherFinBar) {
+            delete this._auxStyles;
+        } else if (otherFinBar instanceof FinBar && otherFinBar['orientation'] !== this.orientation) {
+            var otherStyle = window.getComputedStyle(otherFinBar['bar']),
+                ooh = orientationHashes[otherFinBar['orientation']];
+            this._auxStyles = {};
+            this._auxStyles[whichEnd] = otherStyle[ooh.thickness];
+        }
+        return this; // for chaining
+    },
+
+    /**
+     * @private
+     * @summary Sets the proportional thumb size and hides thumb when 100%.
+     * @desc The thumb size has an absolute minimum of 20 (pixels).
+     * @memberOf FinBar.prototype
+     */
+    _setThumbSize: function () {
+        var oh = this.oh,
+            thumbComp = window.getComputedStyle(this.thumb),
+            thumbMarginLeading = parseInt(thumbComp[oh.marginLeading]),
+            thumbMarginTrailing = parseInt(thumbComp[oh.marginTrailing]),
+            thumbMargins = thumbMarginLeading + thumbMarginTrailing,
+            barSize = this.bar.getBoundingClientRect()[oh.size],
+            thumbSize = Math.max(20, barSize * this.containerSize / this.contentSize);
```

Comment:
Ensures a minimum thumb size of 20px, and sizes the thumb proportional (container size (rows/cols in viewport) vs content size (all rows/cols)